identity/legacy.go

```diff
@@ -35,7 +36,7 @@ func (l *Legacy) Auth(creds *Credentials) (*AuthDetails, error) {
 	details := &AuthDetails{}
 	details.Token = response.Header.Get("X-Auth-Token")
 	if details.Token == "" {
-		return nil, fmt.Errorf("Did not get valid Token from auth request")
+		return nil, gooseerrors.NewUnauthorisedf(nil, "Did not get valid Token from auth request", "")
```

Comment:
The trailing "" here isn't needed. The methods that end in 'f' generally follow the formatting rules of things like Printf,  Sprintf, or Fprintf.  There may be extra args, in this case the first arg being the error cause, then there is the format string, which can include the %s, %v type replacements, then the last args are the values to be formatted into the format string. Here we don't have any formmating values, so the trailing arg here isn't needed.